### ISSUE TITLE
Data table that is sortable should have role button

### ISSUE THREAD

**Issue by Sreeteja33** [primefaces/primevue#7311]
### Describe the bug

For sortable data tables, the headers should have role=”button” for accessibility purposes to indicate to the user that there is an action available to sort the datatable column.

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://primevue.org/datatable/

### Environment

when voice over is on, on sortable column screen reader does not announce that there is action on this column

### Vue version

3.212.0

### PrimeVue version

3.0

### Node version

18.16.0

### Steps to reproduce the behavior

turn on accessibility, navigate to sort column. when user tabs through the column thats sortable screen reader does not announce that there is action, which is sort on column

### Expected behavior

screen reader should announce the action on column header
